### PR TITLE
refactor: allow multiple checkMaterials in contrustor of MaterialMult…

### PR DESCRIPTION
**Pull request by VerschuerenTom**
…iLocationItem

### CODE REVIEW

**Comment by VerschuerenTom** [comment]
Needs testing

File: MiniGamesBox Classic/src/main/java/plugily/projects/minigamesbox/classic/handlers/setup/items/category/MaterialMultiLocationItem.java [hunk 1]
```diff
@@ -249,7 +264,7 @@ private void removeLocation(HumanEntity player, boolean deleteAll) {
           setupInventory.setConfig(keyName, null);
           //considerable to add arena method to remove location
           new MessageBuilder("&e✔ Removed | &a" + name.toUpperCase() + " location for arena " + setupInventory.getArenaKey() + "! (" + location + ")").prefix().send(player);
-          new MessageBuilder("You can now remove the " + checkMaterial.toString() + "!").prefix().send(player);
+          new MessageBuilder("You can now remove the " + checkMaterials.toString() + "!").prefix().send(player);
```
Comment (Tigerpanzer02) [inline MiniGamesBox Classic/src/main/java/plugily/projects/minigamesbox/classic/handlers/setup/items/category/MaterialMultiLocationItem.java:253]: could return unreadable long message for user, getting the "clicked" type would be more accurate

File: MiniGamesBox Classic/src/main/java/plugily/projects/minigamesbox/classic/handlers/setup/items/category/MaterialMultiLocationItem.java [hunk 2]
```diff
@@ -169,25 +184,25 @@ public void onClick(InventoryClickEvent event) {
...
+                new MessageBuilder("&c&l✘ &cPlease only use location where already is a " + checkMaterials + " to add it as a " + name.toUpperCase() + "!").prefix().send(interactEvent.getPlayer());
                 return;
               }
 
               if(location.distance(interactEvent.getClickedBlock().getWorld().getSpawnLocation()) <= Bukkit.getServer().getSpawnRadius()) {
                 new MessageBuilder("&c&l✖ &cWarning | Server spawn protection is set to &6" + Bukkit.getServer().getSpawnRadius()
-                    + " &cand location you want to place is in radius of this protection! &c&lNon opped players won't be able to interact with this " + checkMaterial + " and can't join the game! Reduce the spawn radius (server.properties) or change your location!").prefix().send(interactEvent.getPlayer());
+                        + " &cand location you want to place is in radius of this protection! &c&lNon opped players won't be able to interact with this " + checkMaterials + " and can't join the game! Reduce the spawn radius (server.properties) or change your location!").prefix().send(interactEvent.getPlayer());
```
Comment (Tigerpanzer02) [inline MiniGamesBox Classic/src/main/java/plugily/projects/minigamesbox/classic/handlers/setup/items/category/MaterialMultiLocationItem.java:191]: could return unreadable long message for user, getting the "clicked" type would be more accurate

File: MiniGamesBox Classic/src/main/java/plugily/projects/minigamesbox/classic/handlers/setup/items/category/MaterialMultiLocationItem.java [hunk 3]
```diff
@@ -169,25 +184,25 @@ public void onClick(InventoryClickEvent event) {
...
                 return;
               }
 
               removeLocation(interactEvent.getPlayer(), false);
               break;
             case RIGHT_CLICK_BLOCK:
-              if(block.getType() != checkMaterial) {
-                new MessageBuilder("&c&l✘ &cPlease only use location where already is a " + checkMaterial.toString() + " to add it as a " + name.toUpperCase() + "!").prefix().send(interactEvent.getPlayer());
+              if(!checkMaterials.contains(block.getType())) {
+                new MessageBuilder("&c&l✘ &cPlease only use location where already is a " + checkMaterials + " to add it as a " + name.toUpperCase() + "!").prefix().send(interactEvent.getPlayer());
```
Comment (Tigerpanzer02) [inline MiniGamesBox Classic/src/main/java/plugily/projects/minigamesbox/classic/handlers/setup/items/category/MaterialMultiLocationItem.java:185]: could return unreadable long message for user, getting the "clicked" type would be more accurate

File: MiniGamesBox Classic/src/main/java/plugily/projects/minigamesbox/classic/handlers/setup/items/category/MaterialMultiLocationItem.java [hunk 4]
```diff
@@ -169,25 +184,25 @@ public void onClick(InventoryClickEvent event) {
...
             case PHYSICAL:
             case LEFT_CLICK_AIR:
-              new MessageBuilder("&c&l✘ &cYou can't use a location that is at your player location, please select the " + checkMaterial.toString() + "!").prefix().send(interactEvent.getPlayer());
+              new MessageBuilder("&c&l✘ &cYou can't use a location that is at your player location, please select the " + checkMaterials.toString() + "!").prefix().send(interactEvent.getPlayer());
               break;
             case LEFT_CLICK_BLOCK:
-              if(block.getType() != checkMaterial) {
-                new MessageBuilder("&c&l✘ &cPlease only use location where already is a " + checkMaterial.toString() + " to remove it as a " + name.toUpperCase() + "!").prefix().send(interactEvent.getPlayer());
+              if(!checkMaterials.contains(block.getType())) {
+                new MessageBuilder("&c&l✘ &cPlease only use location where already is a " + checkMaterials+ " to remove it as a " + name.toUpperCase() + "!").prefix().send(interactEvent.getPlayer());
```
Comment (Tigerpanzer02) [inline MiniGamesBox Classic/src/main/java/plugily/projects/minigamesbox/classic/handlers/setup/items/category/MaterialMultiLocationItem.java:177]: could return unreadable long message for user, getting the "clicked" type would be more accurate

File: MiniGamesBox Classic/src/main/java/plugily/projects/minigamesbox/classic/handlers/setup/items/category/MaterialMultiLocationItem.java [hunk 5]
```diff
@@ -169,25 +184,25 @@ public void onClick(InventoryClickEvent event) {
           switch(interactEvent.getAction()) {
             case PHYSICAL:
             case LEFT_CLICK_AIR:
-              new MessageBuilder("&c&l✘ &cYou can't use a location that is at your player location, please select the " + checkMaterial.toString() + "!").prefix().send(interactEvent.getPlayer());
+              new MessageBuilder("&c&l✘ &cYou can't use a location that is at your player location, please select the " + checkMaterials.toString() + "!").prefix().send(interactEvent.getPlayer());
```
Comment (Tigerpanzer02) [inline MiniGamesBox Classic/src/main/java/plugily/projects/minigamesbox/classic/handlers/setup/items/category/MaterialMultiLocationItem.java:173]: could return unreadable long message for user, getting the "clicked" type would be more accurate

File: MiniGamesBox Classic/src/main/java/plugily/projects/minigamesbox/classic/handlers/setup/items/category/MaterialMultiLocationItem.java [hunk 6]
```diff
@@ -129,30 +144,30 @@ public void onClick(InventoryClickEvent event) {
     switch(event.getClick()) {
       case LEFT:
         Block targetBlock = event.getWhoClicked().getTargetBlock(null, 7);
-        if(targetBlock.getType() != checkMaterial) {
-          new MessageBuilder("&c&l✘ &cPlease only look at a location where already is a " + checkMaterial.toString() + " to add it as a " + name.toUpperCase() + "!").prefix().send(event.getWhoClicked());
+        if(!checkMaterials.contains(targetBlock.getType()) ) {
+          new MessageBuilder("&c&l✘ &cPlease only look at a location where already is a " + checkMaterials+ " to add it as a " + name.toUpperCase() + "!").prefix().send(event.getWhoClicked());
```
Comment (Tigerpanzer02) [inline MiniGamesBox Classic/src/main/java/plugily/projects/minigamesbox/classic/handlers/setup/items/category/MaterialMultiLocationItem.java:134]: could return unreadable long message for user, getting the "clicked" type would be more accurate

**Review by Tigerpanzer02 — CHANGES_REQUESTED** [review]
Getting the targetblock type would be enough for the end user about materials ;)

**Comment by Tigerpanzer02** [comment]
@VerschuerenTom Waiting for your adjustments ;)

**Comment by Tigerpanzer02** [comment]
implemented on my own as no respone

**Pull request closed without merging**